proposals/0233-codepoint-and-character-literals.md

```diff
@@ -0,0 +1,234 @@
+
+| `init(`…`)` | `unicodeScalarLiteral` | `extendedGraphemeClusterLiteral` | `stringLiteral` | 
+| --- | --- | --- | --- |
+| `:UInt8`, … , `:Int` | yes* | no  | no  |
+| `:Unicode.Scalar`  | yes  | no  | no  |
+| `:Character`       | yes (upcast) | yes | no  |
+| `:String`          | yes (upcast) | yes (upcast) | yes (upcast) |
+| `:StaticString`    | yes (upcast) | yes (upcast) | yes |
+
+The ASCII range restriction will only apply to single-quote literals coerced to an integer type. Any valid `Unicode.Scalar` can be written as a single-quoted unicode scalar literal, and any valid `Character` can be written as a single-quoted character literal. 
+
+|                    | `'a'` | `'é'` | `'β'` | `'𓀎'` | `'👩‍✈️'` | `"ab"` |
+| --- | --- | --- | --- | --- | --- | --- |
+| `:String`          |          |        |          |       |         | "ab"
+| `:Character`       | `'a'`    | `'é'`  | `'β'`    | `'𓀎'` | `'👩‍✈️'`
+| `:Unicode.Scalar`  | U+0061   | U+00E9 | U+03B2   | U+1300E
+| `:UInt32`          | 97       | 
+| `:UInt16`          | 97       | 
+| `:UInt8`           | 97       | 
+| `:Int8`            | 97       |  
+
+With these changes, the hex code example can be written much more naturally:
+
+```swift
+let hexcodes: [UInt8] = [
+    '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 
+    'a', 'b', 'c', 'd', 'e', 'f'
+]
+
+for scalar in int8buffer {
+    switch scalar {
+    case 'a' ... 'f':
+        // lowercase hex letter
+    case 'A' ... 'F':
+        // uppercase hex letter
+    case '0' ... '9':
+        // hex digit
+    default:
+        // something else
+    }
+}
+```
+
+### Choice of single quotes
+
+We propose to adopt the `'x'` syntax for all textual literal types up to and including `ExtendedGraphemeClusterLiteral`, but not including `StringLiteral`. These literals will be used to express integer types, `Character`, `Unicode.Scalar`, and types like `UTF16.CodeUnit` in the standard library. 
+
+The default inferred literal type for `let x = 'a'` will be `Character`, following the principle of least surprise. This also allows for a natural user-side syntax for differentiating methods overloaded on both `Character` and `String`.
+
+Single-quoted literals will be inferred to be integer types in cases where a `Character` or `Unicode.Scalar` overload does not exist, but an integer overload does. This can lead to strange spellings such as `'1' + 1' == 98`. However, we forsee problems arising from this to be quite rare, as the type system will almost always catch such mistakes, and very few users are likely to express a `String` with two literals instead of the much more obvious `"11"`. 
```

Comment:
Seems like you have an extra ' in this code:

`'1' + 1' == 98`

Reply:
done